### PR TITLE
Writing documentation in four modes

### PR DESCRIPTION
**Pull request by allcaps**
Preview: https://wagtail--6767.org.readthedocs.build/en/6767/explanation/documentation-modes.html

### CODE REVIEW

File: docs/explanation/documentation-modes.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Each type of information delivery has a purpose and targets a specific audience.
+
+  * :ref:`doc-style-tutorial`, learning-oriented
+  * :ref:`doc-style-how-to-guide`, goal-oriented
+  * :ref:`doc-style-reference`, information-oriented
+  * :ref:`doc-style-explanation`, understanding-oriented
+
+Credits `Daniele Procida: What nobody tells you about documentation <https://www.divio.com/blog/documentation/>`__.
+
+Write in style
```
Comment (tomdyson) [inline docs/explanation/documentation-modes.rst:15]:
```suggestion
Choose a writing style
```

File: docs/explanation/documentation-modes.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+--------------
+
+Each page of the Wagtail documentation should be written in **one** mode of information delivery.
+
+Mixing types of information delivery on a single page is bad style.
+These pages are are confusing to read, and harder to understand.
```
Comment (larathompson) [inline docs/explanation/documentation-modes.rst:21]:
```suggestion
These pages are confusing to read, and harder to understand.
```

File: docs/explanation/documentation-modes.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+It is best to split documents that mix the types of information delivery.
+Add links to the first section of each document to cross reference the same topic.
+
+Writing documentation in a specific mode  will help our users to understand, and quickly find what they are looking for.
```
Comment (larathompson) [inline docs/explanation/documentation-modes.rst:25]:
```suggestion
Writing documentation in a specific mode  will help our users to understand and quickly find what they are looking for.
```

File: docs/explanation/documentation-modes.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+`More about tutorials <https://documentation.divio.com/tutorials/>`__
+
+
+.. _doc-style-how-to-guide:
+
+How-to guide
+------------
+
+A guide offer advice on how best to achieve a given task. How-to guides are **goal-oriented**.
```
Comment (larathompson) [inline docs/explanation/documentation-modes.rst:47]:
```suggestion
A guide offering advice on how best to achieve a given task. How-to guides are **goal-oriented**.
```

File: docs/explanation/documentation-modes.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+`More about how-to guides <https://documentation.divio.com/how-to-guides/>`__
+
+
+.. _doc-style-reference:
+
+Reference
+---------
+
+A reference is well structured and allows the reader to find information about a specific
```
Comment (larathompson) [inline docs/explanation/documentation-modes.rst:57]:
```suggestion
A reference is well-structured and allows the reader to find information about a specific
```

File: docs/explanation/documentation-modes.rst [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+`More about reference <https://documentation.divio.com/reference/>`__
+
+
+.. _doc-style-explanation:
+
+Explanation
+-----------
+
+Explanation is high-level and explains concepts and design decisions.
+There little, or no code involved. Background information is **understanding-oriented**.
```
Comment (larathompson) [inline docs/explanation/documentation-modes.rst:69]:
```suggestion
There is little, or no code involved. Background information is **understanding-oriented**.
```

**Comment by cnk** [comment]
This was merged as commit https://github.com/wagtail/wagtail/commit/a4aa5c8cc9641bd19a080f9e7e8b58bb70c5bd98

**Pull request closed without merging**